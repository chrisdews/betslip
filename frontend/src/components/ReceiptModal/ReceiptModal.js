import React from "react";
import PropTypes from "prop-types";

function ReceiptModal({ stakes }) {
  const stakeReducer = (stakes) => {
    const stakeAmounts = Object.values(stakes);
    return stakeAmounts.map((str) => parseFloat(str)).reduce((x, y) => x + y);
  };

  return (
    stakes && (
      <>
        <h1>Receipt</h1>
        <p>Your bet has been placed.</p>
        <div>Total stake: Â£{stakeReducer(stakes)}</div>
      </>
    )
  );
}

ReceiptModal.propTypes = {};

export default ReceiptModal;
