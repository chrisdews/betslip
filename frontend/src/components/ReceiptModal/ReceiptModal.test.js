import React from "react";
import ReceiptModal from "./ReceiptModal";
import { render } from "@testing-library/react";

describe("ReceiptModal", () => {
  it("renders correctly - snapshot", () => {
    const { container } = render(
      <ReceiptModal stakes={{ a: 1, b: 2, c: 3, d: 4 }} />
    );
    expect(container).toMatchSnapshot();
  });
});
