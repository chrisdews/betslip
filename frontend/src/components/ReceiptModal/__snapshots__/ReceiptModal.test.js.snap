// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ReceiptModal renders correctly - snapshot 1`] = `
<div>
  <div
    class="sc-dlnjwi hlOVyA"
  >
    <h1
      class="sc-hKFxyN fzpvuW"
    >
      RECEIPT
    </h1>
    <img
      class="sc-gtsrHT iLDkJB"
      src="oc-logo.png"
    />
    <div
      class="sc-bdnxRM eFtsIP"
    >
      <h3>
        Your bet has been placed.
      </h3>
      <h4>
        Total stake: 
        <b>
          Â£
          10
        </b>
      </h4>
    </div>
  </div>
</div>
`;
